src/repository/LandAndFacilitiesRespository.ts

```diff
@@ -0,0 +1,41 @@
+import { EntityRepository, Repository } from "typeorm";
+import { LandAndFacilities } from "../entity/LandAndFacilities";
+
+@EntityRepository(LandAndFacilities)
+export class LandAndFacilitiesRepository extends Repository<LandAndFacilities> {
+    createLandAndFacilities(
+        country: number,
+        land: number,
+        rigs: number,
+        facs: number,
+        mines: number,
+        isSpawn: boolean,
+        teamControl: number,
+        timestamp: Date
+    ) {
+        const landAndFacilities = new LandAndFacilities();
+        landAndFacilities.country = country;
+        landAndFacilities.land = land;
+        landAndFacilities.rigs = rigs;
+        landAndFacilities.facs = facs;
+        landAndFacilities.mines = mines;
+        landAndFacilities.is_spawn = isSpawn;
+        landAndFacilities.timestamp = timestamp;
+        landAndFacilities.team_control = teamControl;
+        return this.manager.save(landAndFacilities);
+    }
+
+    getSpawnFactories(timestamp: Date) {
+        return this.manager
+            .createQueryBuilder(LandAndFacilities, "landAndFacilities")
+            .select(
```

Comment:
actually its addSelect() i think

Reply:
since you don't actually need the most of the columns, I think you could also do
```ts
    .createQueryBuilder("land_and_facilities")
    .select("country", "facs", "max(timestamp)")
```
instead

Reply:
nvm, that's wrong, I think